src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,238 @@
+                        ? binaryExpression.Right is LiteralExpressionSyntax ? SyntaxKind.GreaterThanToken : SyntaxKind.LessThanToken
+                        : SyntaxKind.EqualsEqualsToken;
+                    negatedExpressionType = binaryExpression.Kind() == SyntaxKind.EqualsExpression
+                        ? binaryExpression.Right is LiteralExpressionSyntax ? SyntaxKind.GreaterThanExpression : SyntaxKind.LessThanExpression
+                        : SyntaxKind.EqualsExpression;
+                }
+
+                result = SyntaxFactory.BinaryExpression(
+                    negatedExpressionType,
+                    binaryExpression.Left,
+                    SyntaxFactory.Token(
+                        binaryExpression.OperatorToken.LeadingTrivia,
+                        negatedOperatorType,
+                        binaryExpression.OperatorToken.TrailingTrivia),
+                    binaryExpression.Right);
+
+                return true;
+            }
+
+            result = null;
+            return false;
+        }
+
+        private ExpressionSyntax Negate(ExpressionSyntax expression, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (TryNegateBinaryComparisonExpression(expression, semanticModel, cancellationToken, out var result))
+            {
+                return result;
+            }
+
+            switch (expression.Kind())
+            {
+                case SyntaxKind.ParenthesizedExpression:
+                {
+                    var parenthesizedExpression = (ParenthesizedExpressionSyntax)expression;
+                    return parenthesizedExpression
+                        .WithExpression(Negate(parenthesizedExpression.Expression, semanticModel, cancellationToken))
+                        .WithAdditionalAnnotations(Simplifier.Annotation);
+                }
+
+                case SyntaxKind.LogicalNotExpression:
+                {
+                    var logicalNotExpression = (PrefixUnaryExpressionSyntax)expression;
+
+                    var notToken = logicalNotExpression.OperatorToken;
+                    var nextToken = logicalNotExpression.Operand.GetFirstToken(
+                        includeZeroWidth: true, includeSkipped: true, includeDirectives: true, includeDocumentationComments: true);
+
+                    var existingTrivia = SyntaxFactory.TriviaList(
+                        notToken.LeadingTrivia.Concat(
+                            notToken.TrailingTrivia.Concat(
+                                nextToken.LeadingTrivia)));
+
+                    var updatedNextToken = nextToken.WithLeadingTrivia(existingTrivia);
+
+                    return logicalNotExpression.Operand.ReplaceToken(
+                        nextToken,
+                        updatedNextToken);
+                }
+
+                case SyntaxKind.LogicalOrExpression:
+                {
```

Comment:
1. nit: consider helper method.
2. please add comment explaining the translation.